.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "third_party/libipt"]
+    path = third_party/libipt
+    url = https://github.com/intel/libipt.git
```

Comment:
I had assumed that libipt was not provided in the main distro repos: but it does seem to be there as `libipt-dev`.  It would be simpler to use that package, just like we do with many other libraries (e.g., for dr$sim compression: libsnappy, lz4, zlib).  Is there some very recent interface that is not part of that package that is the reason to build it from sources?

Reply:
Yes. libipt-dev only contains libipt.so. It can not decode perf recode trace data directly. Currently, drptpostprocessor uses the libipt sideband library (libipt-sb.so) to help handle context switches across images. So we need to build it from the source file. drptpostprocessor is a general solution for perf record data decoder. Using the sideband APIs is a quick solution because we don't need to write more code to handle sideband messages.
But for the data generated by our recorder, the kernel trace instructions may all be decoded from kcore, and we may only need libipt.so. In this case, libipt-dev might meet our requirements.

Reply:
I discussed this question with @abhinav92003 this morning. We think for the current version, there are three reasons that we need to add libipt as a submodule.
- The first reason is what I mentioned above, we need drptpostprocessor to handle trace data recorded by perf, using libipt-sb.so's API is an easy solution. 
- The second reason is except for the image switch, we may need to handle other information like timestamps in the future. 
- The last reason is we may need to handle the image switch when decoding kernel traces.  
So do you agree that we can include it first and if we don't need it we can remove it in the future.

Reply:
OK, seems reasonable.  This is a good reason to isolate all the submodule parts in a separate PR so it can be reverted on its own.